code/game/objects/items/weapons/traps.dm

```diff
@@ -313,11 +313,19 @@ Very rarely it might escape
 		var/mob/living/L = AM
 		if(("\ref[L]" in aware_mobs) && MOVING_DELIBERATELY(L))
 			return ..()
+		prob_catch = initial(prob_catch)
+		if(MOVING_DELIBERATELY(L))
+			prob_catch += - prob_catch*L.stats.getMult(STAT_VIG, STAT_LEVEL_GODLIKE) + 20 * L.get_max_w_class()/ITEM_SIZE_TITANIC
```

Comment:
This check better be a called proc from mob, because not only bear traps must utilize it.